### DISCUSSION TITLE
Azure deployment instructions

### DISCUSSION THREAD

**Discussion by pamelafox** [babybuddy/babybuddy#548] [Show and tell]
Hi everyone!

I'm a mom of a fresh-ish baby (4.5 months) as well as a new advocate at Microsoft Azure. So when I discovered your OSS baby tracker, I thought I'd figure out how to get it deployed to Azure.

I created this repo with the necessary configuration files and instructions:
https://github.com/pamelafox/babybuddy-azure

There are a few ways to host Django apps on Azure, notably App Service and Container Apps, but I went for Container Apps as the pricing is more pay-as-you-go (both have a free tier too, so if it's your only Azure app, it may not matter, but it matters as you deploy more apps). I initially attempted a SQLite DB, but that's not very compatible with the Container Apps model due to the kind of file system used (NFS), so my setup uses a PostGreSQL server, which also has a free tier.

I wanted to share the instructions and also check if it would make sense for me to add the Azure deployment instructions to your official docs anywhere.

Thanks for the great project! Maybe I'll finally understand my baby's nap schedule... :)

**Comment by cdubz** [comment]
Very open to this! I like to support as many platforms as possible.

I tested this as-is but ended up with a failed deploy with the following message:

```json
{
    "status": "Failed",
    "error": {
        "code": "DeploymentFailed",
        "message": "At least one resource deployment operation failed. Please list deployment operations for details. Please see https://aka.ms/DeployOperations for usage details.",
        "details": [
            {
                "code": "Conflict",
                "message": "{\r\n  \"status\": \"Failed\",\r\n  \"error\": {\r\n    \"code\": \"ResourceDeploymentFailure\",\r\n    \"message\": \"The resource operation completed with terminal provisioning state 'Failed'.\",\r\n    \"details\": [\r\n      {\r\n        \"code\": \"InternalServerError\",\r\n        \"message\": \"An unexpected error occured while processing the request. Tracking ID: '2b25ba80-b15d-4fc4-a1f7-d81721bbbe9a'\"\r\n      }\r\n    ]\r\n  }\r\n}"
            }
        ]
    }
}
```

It looks like it was database resource that failed but the error there isn't much more helpful:

```json
{
    "status": "Failed",
    "error": {
        "code": "InternalServerError",
        "message": "An unexpected error occured while processing the request. Tracking ID: '2b25ba80-b15d-4fc4-a1f7-d81721bbbe9a'"
    }
}
```

I tested a redeploy and got the same result.

Regardless, would be happy to get this deploy method added to the docs once we get the kinks worked out!

**Comment by pamelafox** [comment]
Thanks for trying out a deploy! That's a funky error, I haven't seen that particular one before. Was that from the Azure Portal method ("Deploy to Azure" button) or from the Azure CLI method? And was that on an account that already had Azure resources or a new account?
I'll try to replicate myself, but if I can't, I'll poke around to find out who knows what happened. "InternalServerError" sounds like a bug in the Azure codebase.

**Comment by cdubz** [comment]
Portal deploy with a brand new account.

**Comment by pamelafox** [comment]
Sorry for the delayed response! I've been doing a lot of deployment experimentation in the last month and did manage to reproduce an error like that a few times. I removed some constraints from the Postgres server in the infrastructure configuration, and now my deploys seem to always work. Some errors can be location specific, but these locations have worked for me: "West US", "Central US", "East US 2", or "Canada Central".

If you have the time, could you try the "deploy to azure" button again? If it does fail, it'd help to know what parameters you used (especially location).

I also added a video to show the deployment process:
https://www.youtube.com/watch?v=-4ln47lTuRw

**Reply by cdubz** [reply]
No worries. I’ll give this another try soonish. Admittedly I black-holed my Azure account email because the welcome series was way too aggressive.
